src/main/java/pl/kamann/services/email/EmailSender.java

```diff
@@ -13,15 +13,26 @@
 @RequiredArgsConstructor
 public class EmailSender implements EmailSenderFacade {
     private final JavaMailSender javaMailSender;
+    private final EmailContentBuilder emailContentBuilder;
+    private final ResourceBundleEmailMessageProvider messageProvider;
 
     public void sendEmail(String to, String link, Locale userLocale, String type) throws MessagingException {
+        String content = emailContentBuilder.buildConfirmationEmail(type, userLocale, link);
+        sendEmailMessage(new EmailDetails(to, type, userLocale, content));
+    }
+
+    public void sendEmailWithoutConfirmationLink(String to, Locale userLocale, String type) throws MessagingException {
+        String content = emailContentBuilder.buildSampleEmail(type, userLocale);
+        sendEmailMessage(new EmailDetails(to, type, userLocale, content));
+    }
+
+    private void sendEmailMessage(EmailDetails emailDetails) throws MessagingException {
         MimeMessage message = javaMailSender.createMimeMessage();
         MimeMessageHelper helper = new MimeMessageHelper(message, true);
-        EmailContentBuilder emailContentBuilder = new EmailContentBuilder(userLocale, type);
 
-        helper.setTo(to);
-        helper.setSubject(emailContentBuilder.getSubject(type));
-        helper.setText(emailContentBuilder.buildConfirmationEmail(link), true);
+        helper.setTo(emailDetails.getTo());
+        helper.setSubject(messageProvider.getSubject(emailDetails.getType(), emailDetails.getUserLocale()));
+        helper.setText(emailDetails.getContent(), true);
 
```

Comment:
I think we are missing packing MessagingException into dedicated one, it could be also a good idea to setup @Async for sending mails (depend on your wish at the moment :D )